### ISSUE TITLE
[Feature]: Wi-fi only option for foreground backup

### ISSUE THREAD

**Issue by erlenddahl** [immich-app/immich#1377]
### Feature detail

If I understand them correctly, issues #983 and #421 are both closed as the backup options work as intended. Therefore, this is a request for an additional feature: a wi-fi only option for the foreground upload would be very useful. I like the foreground upload, as that gives me a way of controlling when to upload. But when there is no wi-fi only option for it, that also means that I cannot use Immich when I am travelling (without having to rush into settings and deactivating the foreground upload). I think it would be very nice if the wi-fi only option worked for both types of upload.

(I discovered this today, when Immich used 7GB of my data quota because of some assets that failed and were retried a lot of times before I noticed. That may be another issue in itself, of course, but a wifi only option could have saved me.)

### Platform

Mobile App

**Comment by alextran1502** [comment]
@zoodyy Fynn, what do you think about this? I assume we can do this on the Flutter side only right?

**Comment by alextran1502** [comment]
I don't use Android but I think we have this option for Android background backup already

**Comment by alextran1502** [comment]
Ah this is foreground backup.

**Comment by erlenddahl** [comment]
Yep, that's right! I also believe you were right in your first observation, that this is something that must be added on the Flutter side. 

If you guys agree that it would be a good idea to add a wi-fi only option to the foreground backup as well (or a combined switch for foreground/background), I may be able to help. I don't know Flutter yet, but I know programming, so I could give it a try if you want.
